src/stage1/migrate_info.rs

```diff
@@ -43,18 +43,26 @@ pub(crate) struct MigrateInfo {
     work_dir: PathBuf,
     wifis: Vec<WifiConfig>,
     nwmgr_files: Vec<PathBuf>,
+    system_proxy_files: Vec<PathBuf>,
     backup: Option<PathBuf>,
 }
 
 #[allow(dead_code)]
 impl MigrateInfo {
     pub fn new(opts: &Options) -> Result<MigrateInfo> {
         let device = get_device(opts)?;
-        info!("Detected device type: {}", device.get_device_type());
+        let os_name = get_os_name()?;
+        info!("Detected device type: {} running {}", device.get_device_type(), os_name);
 
+        /* If no config.json is passed in command line and we're running on balenaOS,
+         * we can preserve the existing config.json
+         */
         let mut config = if let Some(balena_cfg) = opts.config() {
             BalenaCfgJson::new(balena_cfg)?
-        } else {
+        } else if Path::new("/mnt/boot/config.json").exists() {
```

Comment:
I believe you intended to check also if  `os_name == "balenaOS"` or something like that, given that you added all the infrastructure to read the OS name!

Reply:
Addressed in https://github.com/balena-os/takeover/pull/66/commits/2df5e8c7f21d74732fe1ad71a09e06f2d15dec17#diff-fc1ecdab5d089a49201634bf1ca9669e4f4289a4fbc7f8ac77b144ab883050a1R62